docs/design/logging-integration.adoc

```diff
@@ -0,0 +1,196 @@
+        kind: TaskRun
+      archiveOnDelete: true
+    - selector:
+        apiVersion: v1
+        kind: Pod
+      archiveOnDelete: has(body.metadata.labels["tekton.dev/pipeline"])
+```
+In this case the following happens:
+
+- `PipelineRuns` are archived when they complete.
+- `TaskRuns` are archived when they are deleted.
+- `Pods` are archived when they are deleted and are also part of a Tekton Pipeline
+
+== Generating Log URLs
+
+The logging system is an integration in KubeArchive. All that is required for
+the integration is the successful generation of a URL to access the log for a
+specific `Pod` in the logging system. URLs will ONLY be generated for `Pods`.
+
+To support multiple logging systems, the URLs must be able to be parameterized
+based on the logging system. This could be done via a `ConfigMap` named for the
+logging system being implemented. This `ConfigMap` requires a single entry,
+LOG_URL, whose string value will be evaluated in an environment where variables
+in it will be substituted. For example, take this `ConfigMap` for Splunk:
+```
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kubearchive-splunk
+  namespace: kubearchive
+data:
+  LOG_URL=https://splunk-host/app/search/search?q=search *| spath "kubernetes.pod_id" | search "kubernetes.pod_id"="${POD_ID}"
+  POD_ID=".body.pod_id"
+```
+Any other entry in this `ConfigMap` is treated as a variable, and the value of
+each variable is a JSON path expression that is evaluated against the body of
+the cloud event (i.e. the resource) to determine the real value of that variable
+used in the substitution. For example:
+
+So when generating the logging URL to be stored when a `Pod` is archived,
+the following steps are done:
+
+1. An "environment" is created by evaluating the JSON path expression in each entry
+   in the `ConfigMap`, excluding LOG_URL.
+1. The value for LOG_URL is then evaluated using this environment, resulting in the
+   final URL to the log in the logging system.
+
+== Implementation
+
+=== Database
+
+The KubeArchive database will have a table named `log_url` with three fields:
+
+1. An `id` field as a primary key.
+1. A `resource_id` field which is a foreign key to `resource.id`.  
+1. A `url` field which is the URL for one of of logs.
+
+The `resource_id` field should point back to a `Pod` entry in the `resource` table.
+
+=== Sink
+
+When the sink archives a `Pod`, it must take the additional step go gather all
+the log information and generate the log URL for each. These are stored in the 
+`log_url` table.
+
+The sink should first delete any existing entries in the `log_url` table for the
+`Pod` being archived. Earlier archival requests may have already created records
+in the `log_url` table, and they should be removed to avoid duplicates.
+
+The sink will also need to mount and use the `ConfigMap` for logging as detailed 
+in the <<Configuration>> section.
+
+=== CLI
+
+The CLI will implement a `log` command similar to `kubectl log`. 
```

Comment:
Typo here, the command is `kubectl logs` not `kubectl log`
```suggestion
The CLI will implement a `logs` command similar to `kubectl logs`. 
```

Reply:
Fixed.